taichi/rhi/metal/metal_device.mm

```diff
@@ -403,8 +474,29 @@ DeviceCapabilityConfig collect_metal_device_caps(MTLDevice_id mtl_device) {
   return caps;
 }
 
+std::unique_ptr<MetalSampler> create_sampler(id<MTLDevice> mtl_device) {
+  MTLSamplerDescriptor *desc = [MTLSamplerDescriptor new];
+  desc.magFilter = MTLSamplerMinMagFilterLinear;
+  desc.minFilter = MTLSamplerMinMagFilterLinear;
+  desc.sAddressMode = MTLSamplerAddressModeMirrorRepeat;
+  desc.tAddressMode = MTLSamplerAddressModeMirrorRepeat;
+  desc.rAddressMode = MTLSamplerAddressModeMirrorRepeat;
+  desc.compareFunction = MTLCompareFunctionAlways;
+  desc.mipFilter = MTLSamplerMipFilterLinear;
+
+  id<MTLSamplerState> mtl_sampler_state =
+      [mtl_device newSamplerStateWithDescriptor:desc];
+
+  [desc release];
+
+  return std::make_unique<MetalSampler>(mtl_sampler_state);
+}
+
 MetalDevice::MetalDevice(MTLDevice_id mtl_device) : mtl_device_(mtl_device) {
   compute_stream_ = std::unique_ptr<MetalStream>(MetalStream::create(*this));
+  graphics_stream_ = std::unique_ptr<MetalStream>(MetalStream::create(*this));
```

Comment:
Has the `semaphore` between streams been implemented yet? Without that true-multi-stream will break.

Reply:
oof, i'll reuse compute_stream then